### ISSUE TITLE
PHP strict error when running behat. 

### ISSUE THREAD

**Issue by Jodes81** [Behat/Behat#773]
I have configured Behat to use the PHP binary 5.4.12 installed using WAMP. 

When I run Behat, it produces the following error (then followed by the features, scenarios and results).

```
Strict standards: Declaration of Behat\Behat\Console\Input\InputDefinition::getSynopsis() should be compatible with Symfony\Component\Console\Input\InputDefinition::getSynopsis($short = false) in C:\phpbin\behat\vendor\behat\behat\src\Behat\Behat\Console\Input\InputDefinition.php on line 157

Call Stack:
    0.0003     238128   1. {main}() C:\phpbin\behat\vendor\behat\behat\bin\behat:0
    0.0282    1813112   2. Symfony\Component\Console\Application->run() C:\phpbin\behat\vendor\behat\behat\bin\behat:32
    0.0399    2201712   3. Behat\Behat\Console\BehatApplication->doRun() C:\phpbin\behat\vendor\symfony\console\Application.php:126
    0.0399    2201776   4. Behat\Behat\Console\BehatApplication->createCommand() C:\phpbin\behat\vendor\behat\behat\src\Behat\Behat\Console\BehatApplication.php:66
    0.2908    6867392   5. Symfony\Component\DependencyInjection\ContainerBuilder->get() C:\phpbin\behat\vendor\behat\behat\src\Behat\Behat\Console\BehatApplication.php:80
    0.2909    6868528   6. Symfony\Component\DependencyInjection\ContainerBuilder->createService() C:\phpbin\behat\vendor\symfony\dependency-injection\ContainerBuilder.php:504
    0.3100    7223848   7. ReflectionClass->newInstanceArgs() C:\phpbin\behat\vendor\symfony\dependency-injection\ContainerBuilder.php:980
    0.3100    7225632   8. Behat\Behat\Console\Command\BehatCommand->__construct() C:\phpbin\behat\vendor\symfony\dependency-injection\ContainerBuilder.php:980
    0.3101    7226760   9. Composer\Autoload\ClassLoader->loadClass() C:\phpbin\behat\vendor\symfony\dependency-injection\ContainerBuilder.php:0
    0.3107    7226912  10. Composer\Autoload\includeFile() C:\phpbin\behat\vendor\composer\ClassLoader.php:301
```
This is a problem because it severely impacts the clarity of the test results, and crucially, may mask other errors if a user/developer becomes php-error-blind so to speak. 

The only workaround I have left to try is to add $short = false as per this declaration: Behat\Behat\Console\Input\InputDefinition::getSynopsis($short = false). However, this would get overwritten each time the setup is updated (using composer).

I have tried editing php.ini and setting error_reporting to  E_ALL & ~E_STRICT, but it makes no difference. 

I would write a text manipulation script to remove the above error from the output, but it would be brittle to say the least.

The above error and workarounds suggested I have taken from [my question on stackoverflow.com](http://stackoverflow.com/questions/31643189/running-behat-outputs-annoying-php-strict-standards-error-notice)

**Comment by stof** [comment]
Can you show the version of the console component you are using with ``composer show symfony/console -i`` (or ``composer show symfony/symfony -i`` if the previous command tells you it is not found) ?

**Comment by stof** [comment]
and are you using Behat 2.5 or 3 ?

**Comment by Jodes81** [comment]
I am not sure what version of Behat I'm using! But my composer.json is as follows:
```
{
    "require": {
        "behat/behat": "2.4.*@stable",
        "behat/mink": "1.4.*@stable",
        "behat/mink-extension": "*",
        "behat/mink-goutte-driver": "*",
        "behat/mink-selenium2-driver": "*"
    },
    "minimum-stability": "dev",
    "config": {
        "bin-dir": "bin/"
    }
}
```
And when I run `Behat --version`, it returns `Behat version DEV`. (Which I don't understand - my composer.json implies it should be using version 2.4.*).

When I run `composer show symfony/console -i` I get the following:
```
name     : symfony/console
descrip. : Symfony Console Component
keywords :
versions : * 2.8.x-dev
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
source   : [git] https://github.com/symfony/Console.git 575782770a015042d2fe34f1388d5073bde784bd
dist     : [zip] https://api.github.com/repos/symfony/Console/zipball/575782770a015042d2fe34f1388d5073bde784bd 575782770a015042d2fe34f1388d5073bde784bd
names    : symfony/console

autoload
psr-4
Symfony\Component\Console\ => .

requires
php >=5.3.9

requires (dev)
psr/log ~1.0
symfony/event-dispatcher ~2.1|~3.0.0
symfony/phpunit-bridge ~2.7|~3.0.0
symfony/process ~2.1|~3.0.0

suggests
psr/log For using the console logger
symfony/event-dispatcher
symfony/process
```

**Comment by ciaranmcnulty** [comment]
Your `minimum-stability: dev` will be overriding any package-specific stability levels, so you're getting a 2.x dev version.

**Comment by Jodes81** [comment]
Ugh - I don't know how I missed that. But removing `minimum-stability: dev` then running `composer update` and `behat --version` still outputs `Behat version DEV`

**Comment by Jodes81** [comment]
Apologies, I missed that. However, when I remove `minimum-stability: dev`, run `composer update` and `behat --version` I still get `Behat version DEV`. Is this the reason for the PHP errors?

**Comment by stof** [comment]
2.4 is several years old and it is unmaintained. Please use either 2.5 or 3.x

**Comment by stof** [comment]
use ``composer show behat/behat -i`` to see the installed version of behat (or look in your composer.lock)

**Comment by stof** [comment]
Btw, I'm closing this issue because it was already fixed in Behat 2.5
